src/main/java/com/epam/lab/hospitalspring/controller/PatientController.java

```diff
@@ -0,0 +1,129 @@
+    PatientService patientService;
+    @Autowired
+    DiagnosisService diagnosisService;
+    @Autowired
+    PrescriptionService prescriptionService;
+
+    LocalDateTime today = LocalDateTime.now();
+
+    @GetMapping(value = "/patients")
+    public String getPatients(Model model, Authentication authentication) {
+        PersonalDetailsImpl personalDetailsService = (PersonalDetailsImpl) authentication.getPrincipal();
+        Role currentRole = personalDetailsService.getPersonal().getRole();
+        if (currentRole != Role.ADMIN) {
+            model.addAttribute("patients", patientService.getNotDeletedPatients());
+            model.addAttribute("currentRole", personalDetailsService.getPersonal().getRole());
+            model.addAttribute("firstName", personalDetailsService.getPersonal().getFirstName());
+            model.addAttribute("lastName", personalDetailsService.getPersonal().getLastName());
+            return "patients";
+        } else {
+            return "redirect:/error/errorMessage";
+        }
+    }
+
+    @GetMapping(value = "/deletedPatients")
+    public String getDeletedPatients(Model model, Authentication authentication) {
+        PersonalDetailsImpl personalDetailsService = (PersonalDetailsImpl) authentication.getPrincipal();
+        Role currentRole = personalDetailsService.getPersonal().getRole();
+        model.addAttribute("patients", patientService.getDeletedPatients());
+        if (currentRole == Role.DOCTOR) {
+            model.addAttribute("currentRole", personalDetailsService.getPersonal().getRole());
+            model.addAttribute("firstName", personalDetailsService.getPersonal().getFirstName());
+            model.addAttribute("lastName", personalDetailsService.getPersonal().getLastName());
+            return "deletedPatients";
+        } else {
+            return "redirect:/error/errorMessage";
+        }
+    }
+
+    @GetMapping(value = "/patientDiagnosisCard/{id}")
+    public String getPatientDiagnosisCard(@PathVariable("id") Long id,
+                                          Model model, Authentication authentication, Locale locale) {
+        PersonalDetailsImpl personalDetailsService = (PersonalDetailsImpl) authentication.getPrincipal();
+        Role currentRole = personalDetailsService.getPersonal().getRole();
+
+        model.addAttribute("role", currentRole);
+        if (currentRole != Role.ADMIN) {
+            Patient currentPatient = patientService.getPatientById(id);
+            model.addAttribute("patient", currentPatient);
+            Gson gson = GsonFactory.buildGson(locale);
+            model.addAttribute("diagnoses", gson.toJson(currentPatient.getDiagnosisList()));
+            return "patientDiagnosisCard";
+        } else {
+            return "redirect:/error/errorMessage";
+        }
+    }
+
+    @GetMapping(value = "/addPatient")
+    public String getAddPatientPage(Patient patient) {
+        return "patient";
+    }
+
+    @PostMapping(value = "/addPatient")
```

Comment:
Don't use verb in uri if you follow rest api. It should be something like written below
localhost / patients - post if you create, put for update, get for getting, delete for deleting